arviz/data/inference_data.py

```diff
@@ -178,9 +178,23 @@ def _groups_all(self):
         return self._groups + self._groups_warmup
 
     def __iter__(self):
-        for group in self._groups_all:  # pylint: disable=protected-access
-            dataset = getattr(self, group)
-            yield group, dataset
+        for group in self._groups_all:
+            yield group
+
+    def groups(self):
+        return self._groups_all
+
+    def values(self):
+        datasets = []
+        for group in self._groups_all:
+            datasets.append(getattr(self, group))
+        return datasets
```

Comment:
for `groups` a list is fine because we already have that same list. here however, I think using generators will perform better

```suggestion
        for group in self._groups_all:
            yield getattr(self, group)
```